compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -81,6 +81,9 @@ case class Settings(
   def withSave: Settings =
     this.copy(save = true)
 
+  def noSave: Settings =
+    this.copy(save = false)
+
```

Comment:
Why do we need this? There is `save = false` by default afaik

Reply:
> Why do we need this? There is `save = false` by default afaik

The purpose is to permit a script to opt-out of the `-save` option if necessary, without having to disable the `-save` option for all scripts (by removing `-save` from `SCALA_OPTS`).   There is a new command line option `-nosave` that will reset `save = false` after `SCALA_OPTS` have been processed.   The new option can be added to the hashbang line.   Our new `BashScriptTests` test case verifies that it can override the `-save` option.